### DISCUSSION TITLE
Compact option for messages

### DISCUSSION THREAD

**Discussion by ghost** [sakura-ryoko/afkplus#27] [Ideas]
Instead of sending two messages for AFK and PROTECTION status, be able to define just one message.

![image](https://github.com/sakura-ryoko/afkplus/assets/120551916/71cf5d03-cd98-4cec-930c-c63fc0d2cc81)

**Comment by sakura-ryoko** [comment]
The way the mod is coded currently, it's not quite straight forward, because they are totally separate functions with separate hooks.  In the future, I do plan on re-writing a lot of the mod with a more modern coding style / interface / config format, which could make that sort of thing a bit easier.

The Invulnerable function was added as a response to a suggestion, which is an optional feature; and the timing is not the same due to the Delay of it activating.
With the Deactivation, since they happen at the same time it would be a bit easier.

Also, if the whenAFK and whenDisableDamage configs are set different ways, with an AFK reason; what would be a good approach to this ?  The string could get quite long.

So, I understand the suggestion, but I think that I would mark it as for a future update, because in order to change how it currently functions, it would need lot's of re-thinking and re-working of existing code.

**Comment by sakura-ryoko** [comment]
And to be clear, I do have plans for doing somewhat of a mod re-write / overhaul in the coming future; hopefully sometime this year; and I have another coder who can potentially give me a hand with the project; and I hope to introduce new concepts; such as making my AFK mod into more like an API.

**Comment by sakura-ryoko** [comment]
I have moved this [Issue#26](https://github.com/sakura-ryoko/afkplus/issues/26) to a Discussion.

**Comment by ghost** [comment]
Sounds good! Thanks.

**Comment by ghost** [comment]
Any update on this? 😄
